File: packages/observability/nextjs/sentry.ts
export async function initSentry(dsn?: string, environment?: string) {
    const resolvedDsn = dsn || process.env.SENTRY_DSN
    if (!resolvedDsn) {
        return
    }

    const Sentry = await import("@sentry/nextjs")
    Sentry.init({
        dsn: resolvedDsn,
        environment: environment || process.env.NEXT_PUBLIC_APP_ENV || "production",
        tracesSampleRate: 0.1,
    })
}

File: packages/observability/python/crewcircle_observability/sentry.py
import os


def init_sentry(dsn=None, environment=None, sample_rate=0.1):
    dsn = dsn or os.getenv("SENTRY_DSN")
    if not dsn:
        return

    import sentry_sdk

    sentry_sdk.init(
        dsn=dsn,
        environment=environment or os.getenv("APP_ENV", "production"),
        traces_sample_rate=sample_rate,
    )
